### PR TITLE
Adding docker-compose and the beginning of the related documentation

### PR DESCRIPTION
**Pull request by EkielZan**
To ease the setup of Storywise through docker, I've created a docker-compose.yml file.
Here is it with a small MD file as documentation. I'll work on a more advanced version of the compose file.

One note though. Due to my hardware limitation (CPU doesn't support AVX ), I had to use the version 4.4.18 of MongoDB. 
 (u/Imarite on reddit)

### CODE REVIEW

File: docker/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - DATABASE_NAME=storywise
+      - USERNAME=${USERNAME}
+      - PASSWORD=${PASSWORD}
+      - API_BASE_URL=${API_BASE_URL}
+    networks:
+      - storywise
+    depends_on:
+      - mongodb
+    ports:
+      - 9280:3000
```
Comment (MihaiNueleanu) [inline docker/docker-compose.yaml:22]: could the port be parametrised via env variable, with a default to 3000?

File: docker/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
+version: "3"
+services:
+  storywise:
+    image: mihainueleanu/storywise:latest
+    container_name: storywise
+    hostname: storywise
+    environment:
+      - PUID=1000
+      - PGID=1000
```
Comment (MihaiNueleanu) [inline docker/docker-compose.yaml:10]: Interesting. I only use docker for running local workloads (for development purposes, and for hosting things I use kubernetes), and this is the first time I'm seeing PUID, PGID being set in a docker-compose. Do you happen to have some article as to why this is good practice in docker? I'm just curious to know

File: docker/docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      - API_BASE_URL=${API_BASE_URL}
+    networks:
+      - storywise
+    depends_on:
+      - mongodb
+    ports:
+      - 9280:3000
+    restart: unless-stopped
+  mongodb:
+    image: mongo:4.4.18
```
Comment (MihaiNueleanu) [inline docker/docker-compose.yaml:25]: could we upgrade to a newer mongo version ?

File: docker/docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      - MONGO_INITDB_ROOT_PASSWORD=${MONGO_INITDB_ROOT_PASSWORD}
+      - MONGO_INITDB_DATABASE=${DATABASE_NAME}
+    ports:
+      - 27017:27017
+    volumes:
+      - /data/mongodb:/data/db
+    restart: unless-stopped
+networks:
+  storywise:
+    external: true
```
Comment (MihaiNueleanu) [inline docker/docker-compose.yaml:42]: I haven't thought this through fully, but I'm just thinking - can the network be avoided altogether? I'm thinking it's a bit simpler if one doesn't have the network setup at all

File: docker/docker-compose.yaml (hunk 1 above)
Comment (EkielZan) [reply docker/docker-compose.yaml:22]: I can check that.. I will look at that asap. 
Also, fyi, I've adapted also the Dockerfile on my side to build a ARM compatible image. If you're interested. I can add it as well.

**Comment by EkielZan** [comment]
Got lots of late work to do so I'm a bit late to answer. I'll respond to all question here:

First the special network, I use it because the default "bridge" network doesn't support DNS ( See [Here](https://stackoverflow.com/questions/41400603/dockers-embedded-dns-on-the-default-bridged-network) ).
So to ease the connection between the containers, I'm using user created network.
Just do `docker network create storywise`  ( or whatever name you want but update the yaml accordingly) 

for the version of mongodb, you can indeed use the latest version, I couldn't test with the new version because my homelad CPU doesn't support AVX and thus is not compatible with this version. In fact, I'm currently testing storywise on a VPS with the latest version and it seems to work fine.

For the PUID/GUID, it's mainly to be able to read/write in the local created file with my linux user.  I will remove it from the compose file because it can break some containers.

**Comment by EkielZan** [comment]
To add on the network topic.
You can use the default bridge and use the "--link" to link 2 containers but it's deprecated and most likely will be removed soon.

https://docs.docker.com/network/links/

**Review by MihaiNueleanu — APPROVED** [review]

**Comment by MihaiNueleanu** [comment]
thank you for the first PR! @EkielZan

**Pull request merged**